### PR TITLE
Add support for compute_kzg_proof

### LINKED ISSUE #18: Cryptol Spec: `compute_quotient_eval_within_domain`
Goal: implement the Cryptol function for [`compute_quotient_eval_within_domain `](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_quotient_eval_within_domain)

Task(s):
- [x] Create `KZG.cry`
- [x] Implement `compute_quotient_eval_within_domain ` in `KZG.cry`

Dependencies (same as #17):
- [x] Implement [`compute_roots_of_unity`](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_roots_of_unity) in `BlsHelpers.cry`

### LINKED ISSUE #19: Cryptol Spec: `compute_kzg_proof`
Goal: implement the Cryptol function for [`compute_kzg_proof `](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_kzg_proof)

Task(s):
- [x] Implement `compute_kzg_proof` in `KZG.cry`

Dependencies:
- [x] Implement [`blob_to_polynomial`](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#blob_to_polynomial) in `BlsHelpers.cry`
- [x] Implement [`bytes_to_bls_field `](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#bytes_to_bls_field) in `BlsHelpers.cry`
- [x] Implement [`compute_kzg_proof_impl`](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_kzg_proof_impl) in `KZG.cry`

### PR DESCRIPTION
**Pull request by b13decker**
Here are the next functions that allow us to compute KZG proofs. There is a slight detour as we tried to speed up the scalar multiplication of G1 points (with some success).

Closes #18, closes #19

### CODE REVIEW

File: spec/Spec/BlsHelpers.cry [hunk 1]
```diff
@@ -66,9 +67,9 @@ compute_powers x
  * NOTE: the Python multi_exp handles both G1 and G2 points, and just calls their
  *  corresponding `add` and `multiply` functions. This implementation is for G1 points only.
  */
-g1_multi_exp : {len} (fin len, len >= 1) => [len]G1Point -> [len]UInt256 -> G1Point
+g1_multi_exp : {len} (fin len, len >= 1, 256 >= width (len - 1)) => [len]G1Point -> [len]UInt256 -> G1Point
 g1_multi_exp points scalars = results!0 where
-    results = [G1_INFINITY] # [g1_add (results@i) (g1_multi (points@i) (scalars@i)) | i <- [0 .. len-1]]
+    results = [G1_INFINITY] # [g1_add r (g1_multi_with_table p s i) | r <- results | p <- points | s <- scalars | i <- [0 .. (len-1)]]
```
Comment (mccleeary-galois) [inline spec/Spec/BlsHelpers.cry:72]: We might want to consider having to option to run the original "to the spec" multiplication as well.

Comment (b13decker) [reply spec/Spec/BlsHelpers.cry]: Pushed some new commits to address this.

**Comment by mccleeary-galois** [comment]
We should consider if we want to keep the table implementation the performance improvements did exist but were pretty modest.

**Review by mccleeary-galois — APPROVED** [review]

**Comment by b13decker** [comment]
@mccleeary-galois In the end, I don't think the table version is work keeping. I've dropped those commits and just kept the original ones.

**Pull request merged**